### ISSUE TITLE
Core panic if deinitializing the second time

### ISSUE THREAD

**Issue by Mr-Mime** [h2zero/NimBLE-Arduino#303]
# Reproduce
Steps to reproduce:
1. init the BLE Device
2. deinit the BLE Device 
3. init the BLE Device
4. deinit the BLE Device

We discovered this bug in a bigger project, but I was able to recreate it in an simple application.
We have the `test.ino` which looks like this:

<details>
  <summary>test.ino</summary>

```
#include "log.h"
#include "ble.h"

void setup(void) {
  log_init();

  ble_init();
  delay(3000);

  ble_deinit();
  delay(3000);

  ble_init();
  delay(3000);

  ble_deinit();
}

void loop(void) {
  log_print("Entered loop");
  delay(3000);
}
```

</details>

The steps of the `ble.cpp` are based on the examples [here](https://h2zero.github.io/esp-nimble-cpp/md_docs__new_user_guide.html#creating-a-server) and the content looks like this:
<details>
  <summary>ble.cpp</summary>

```
#include <NimBLEDevice.h>
#include <NimBLEServer.h>

#include "log.h"
#include "ble.h"

void ble_init(void) {
  NimBLEDevice::init("TEST");

  NimBLEServer *pServer = NimBLEDevice::createServer();
  NimBLEService *pService = pServer->createService("ABCD");
  NimBLECharacteristic *pCharacteristic = pService->createCharacteristic("1234");

  pService->start();
  pCharacteristic->setValue("Hello BLE");

  NimBLEAdvertising *pAdvertising = NimBLEDevice::getAdvertising();
  pAdvertising->addServiceUUID("ABCD");
  pAdvertising->start();
}

void ble_deinit(void) {
  NimBLEDevice::deinit(true);
}
```
</details>

# Debugging
I used [EspArduinoExceptionDecoder](https://github.com/me21/EspArduinoExceptionDecoder) to get some information from the backtrace...

When using `deinit(true)` I get the following:
```
stack:
0x4008ebdc: invoke_abort at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/esp32/panic.c:156
0x4008ee59: abort at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/esp32/panic.c:171
0x40083bf9: lock_acquire_generic at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/newlib/locks.c:143
0x40083d25: _lock_acquire_recursive at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/newlib/locks.c:171
0x400f4e83: _vfiprintf_r at /Users/ivan/e/newlib_xtensa-2.2.0-bin/newlib_xtensa-2.2.0/xtensa-esp32-elf/newlib/libc/stdio/../../../.././newlib/libc/stdio/vfprintf.c:860 (discriminator 2)
0x400ee9e9: fiprintf at /Users/ivan/e/newlib_xtensa-2.2.0-bin/newlib_xtensa-2.2.0/xtensa-esp32-elf/newlib/libc/stdio/../../../.././newlib/libc/stdio/fiprintf.c:50
0x400d8e14: __assert_func at /Users/ivan/e/newlib_xtensa-2.2.0-bin/newlib_xtensa-2.2.0/xtensa-esp32-elf/newlib/libc/stdlib/../../../.././newlib/libc/stdlib/assert.c:59 (discriminator 8)
0x40093235: multi_heap_free at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/heap/multi_heap_poisoning.c:218 (discriminator 1)
0x40085c6e: heap_caps_free at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/heap/heap_caps.c:268
0x40083d7d: _free_r at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/newlib/syscalls.c:42
0x400d94b1: operator delete(void*) at /builds/idf/crosstool-NG/.build/src/gcc-5.2.0/libstdc++-v3/libsupc++/del_op.cc:46
0x400dbc97: NimBLEDevice::deinit(bool) at /home/lennox/devel/ESP/test/lib/NimBLE/NimBLEDevice.cpp:894 (discriminator 1)
0x400dab21: ble_deinit() at /home/lennox/devel/ESP/test/src/ble.cpp:25
0x400da9ed: setup() at /home/lennox/devel/ESP/test/src/test.ino:12
0x400da1e6: loopTask(void*) at /home/lennox/devel/ESP/Arduino-ESP32-1.0.6/cores/esp32/main.cpp:18
0x4008fe6a: vPortTaskWrapper at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/freertos/port.c:143
```

When using `deinit(false)` I get the following:
```
stack:
0x400924c5: uxListRemove at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/freertos/list.c:214
0x4009200b: prvProcessReceivedCommands at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/freertos/timers.c:737
0x400920db: prvTimerTask at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/freertos/timers.c:547 (discriminator 1)
0x4008fe6a: vPortTaskWrapper at /home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/freertos/port.c:143
```

At this moment I don't know where to dig deeper to.
If I can help somehow, I am happy to do so, just let me know.

**Comment by h2zero** [comment]
Thanks for reporting, does this happen on the master branch or is this on the 1.3.1 release?

**Comment by h2zero** [comment]
Sorry for the delay, this is fixed now in d0d9edd
